src/test/java/com/commercetools/sync/sdk2/inventories/helpers/InventoryReferenceResolverTest.java

```diff
@@ -0,0 +1,290 @@
+              assertThat(exception.getCause().getMessage()).isEqualTo(expectedMessageWithCause);
+              return null;
+            })
+        .toCompletableFuture()
+        .join();
+  }
+
+  @Test
+  void
+      resolveSupplyChannelReference_WithEmptyIdOnSupplyChannelReference_ShouldNotResolveChannelReference() {
+    final InventoryEntryDraft draft =
+        InventoryEntryDraftBuilder.of()
+            .sku(SKU)
+            .quantityOnStock(QUANTITY)
+            .expectedDelivery(DATE_1)
+            .restockableInDays(RESTOCKABLE_IN_DAYS)
+            .supplyChannel(ChannelResourceIdentifierBuilder.of().key("").build())
+            .custom(CUSTOM_FIELDS_DRAFT)
+            .build();
+
+    final InventoryReferenceResolver referenceResolver =
+        new InventoryReferenceResolver(syncOptions, typeService, channelService);
+
+    referenceResolver
+        .resolveSupplyChannelReference(InventoryEntryDraftBuilder.of(draft))
+        .exceptionally(
+            exception -> {
+              assertThat(exception).isExactlyInstanceOf(ReferenceResolutionException.class);
+              assertThat(exception.getMessage())
+                  .isEqualTo(
+                      format(
+                          "Failed to resolve supply channel resource identifier on "
+                              + "InventoryEntryDraft with SKU:'1000'. Reason: %s",
+                          BLANK_KEY_VALUE_ON_RESOURCE_IDENTIFIER));
+              return null;
+            })
+        .toCompletableFuture()
+        .join();
+  }
+
+  @Test
+  void
+      resolveSupplyChannelReference_WithNullIdOnChannelReference_ShouldNotResolveSupplyChannelReference() {
+    final InventoryEntryDraft draft = mock(InventoryEntryDraft.class);
+    final ChannelResourceIdentifier supplyChannelReference =
+        ChannelResourceIdentifierBuilder.of().id(null).build();
+    when(draft.getSupplyChannel()).thenReturn(supplyChannelReference);
+
+    final InventoryReferenceResolver referenceResolver =
+        new InventoryReferenceResolver(syncOptions, typeService, channelService);
+
+    referenceResolver
+        .resolveSupplyChannelReference(InventoryEntryDraftBuilder.of(draft))
+        .exceptionally(
+            exception -> {
+              assertThat(exception).isExactlyInstanceOf(ReferenceResolutionException.class);
+              assertThat(exception.getMessage())
+                  .isEqualTo(
+                      format(
+                          "Failed to resolve supply channel resource identifier on "
+                              + "InventoryEntryDraft with SKU:'null'. Reason: %s",
+                          BLANK_KEY_VALUE_ON_RESOURCE_IDENTIFIER));
+              return null;
+            })
+        .toCompletableFuture()
+        .join();
+  }
+
+  @Test
+  void
+      resolveSupplyChannelReference_WithResolvedSupplyChannelReference_ShouldNotResolveChannelReference() {
```

Comment:
How do you verify that the test does not resolve channel reference? It could resolve channel reference as well as the assertions are exactly the same as for `resolveSupplyChannelReference_WithNonExistingChannelAndEnsureChannel_ShouldResolveSupplyChannelReference`

Reply:
You're right, this test doesn't have value. I removed it: [2750321](https://github.com/commercetools/commercetools-sync-java/pull/1023/commits/2750321bafaad23b63e0a69ef62cf5f03281265b)